src/boot/CIC6105.c

```diff
@@ -2,23 +2,65 @@
 
 #include "build.h"
 #include "fault.h"
+#include "macros.h"
+#include "regs.h"
+#include "scheduler.h"
 
-#pragma increment_block_number "n64-us:200"
+#pragma increment_block_number "n64-us:188"
+
+// TODO: Consider moving to a header file.
+extern u64 cic6105TextStart[];
+extern u64 cic6105TextEnd[];
 
 s32 gCICAddr1Val;
 s32 gCICAddr2Val;
 FaultClient sRomInfoFaultClient;
+#if MM_VERSION <= N64_JP_1_1
+s32 gCICValue1;
+s32 gCICValue2;
+
+s32 CIC6105_ILoveYou(void);
+#endif
 
 void CIC6105_Noop1(void) {
+#if MM_VERSION < N64_US
+    SREG(20) = 1;
+#endif
 }
 
 void CIC6105_Noop2(void) {
+#if MM_VERSION < N64_US
+    SREG(20) = 0;
+#endif
 }
 
 void CIC6105_PrintRomInfo(void) {
+#if MM_VERSION >= N64_US
     FaultDrawer_DrawText(80, 200, "SP_STATUS %08x", IO_READ(SP_STATUS_REG));
     FaultDrawer_DrawText(40, 184, "ROM_F [Creator:%s]", gBuildCreator);
     FaultDrawer_DrawText(56, 192, "[Date:%s]", gBuildDate);
+#else
+    u32 status = IO_READ(SP_STATUS_REG);
+
+    FaultDrawer_SetCursor(48, 200);
+    if (status & SP_STATUS_SIG7) {
+        FaultDrawer_Printf("OCARINA %08x %08x", gCICValue1, gCICValue2);
+    } else {
+        FaultDrawer_Printf("LEGEND %08x %08x", gCICValue1, gCICValue2);
+    }
+    FaultDrawer_SetCursor(40, 184);
+    FaultDrawer_Printf("ROM_F");
+    FaultDrawer_Printf(" [Creator:%s]", gBuildCreator);
+    FaultDrawer_SetCursor(56, 192);
+    FaultDrawer_Printf("[Date:%s]", gBuildDate);
+    FaultDrawer_SetCursor(96, 32);
+    FaultDrawer_Printf("I LOVE YOU %08x", CIC6105_ILoveYou());
+
+    {
+        // TODO: where to put this?
+        static const char unused[] = "flag:%08x data:%08x\n";
```

Comment:
Personally I think I would put it in the ScheduleCICTask function, though is fine if you want to keep it where it is now.
If you do put it in a function though you should be able to do
`(void)"flag:%08x data:%08x\n";` and it should work fine. No need for the extra scope.